1) Since `i` was initialized using `var`, this means that `i` does not have block scope. As a result, even though `i` is initalized within the for loop, it can still be accessed outside of that for loop in the function (function-scope). So, the value of `i` will be printed at Line 11, which is representative of the length of `prices` - 1.
2) Since `discountedPrice` was initalized using `var` within the for loop, this means that `discountedPrice` does not have block scope. As a result, `discountedPrice` can be accessed outside of the for loop within the function. As a result, Line 12 will print the discounted price of the last price element in the `prices` parameter. This value is equal to prices[prices.length - 1] * (1 - discount).
3) Since `finalPrice` is within the scope of Line 13, as they are both under the function `discountPrices`, it's value is able to be printed at Line 13. This value will take the same value as that printed by Line 12, however it will be rounded to the nearest 2 decimal places (i.e. 12.75).
4) If we call `discountPrices([100, 200, 300], .5)`, the function will return the array `discounted` whose value is `[50, 100, 150]`. To explain, the fucntion will traverse through the parameter `prices` array and within the for loop, it will take every element in `prices` and calculate the discounted price and round that discounted price to 2 decimal places. Then, these calculated discounted prices are pushed into a `discounted` array, which will then be returned by the function. Thus, giving us `[50, 100, 150]` for the call `discountPrices([100, 200, 300], .5)`.
5) At Line 11, a reference error will occur because `i` was initialized using `let` in a for loop block. As a result, the scope of `i` is confined to the for loop block and cannot be accessed outside of it. So, at Line 11 the reference error will state that it cannot find the variable `i`.
6) At Line 12, a reference error will occur because `discountedPrice` was initialized using `let` in a for loop block. As a result, the scope of `discountedPrice` is confined to the for loop block and cannot be accessed outside of it. So, at Line 12 the reference error will state that it cannot find the variable `discountedPrice`.
7) At Line 13, the value of finalPrice will be printed, this value will be equal to the last price element in `prices` with the discount and 2-decimal rounding applied to it. This is because finalPrice was initialized using `let` and its scope is the entirity of the function `discountPrices` which means it is accessible at line 13, as line 13 is within the function `discountPrices`.
8) If the scope errors given by lines 11 and 12 are not resolved, the function will not run and nothing would be returned. If the lines (11,12) that are giving scope errors are removed, then the function on the call `discountPrices([100, 200, 300], .5)` will return the array `discounted` whose value would be `[50, 100, 150]`.To explain, the fucntion will traverse through the parameter `prices` array and within the for loop, it will take every element in `prices` and calculate the discounted price and round that discounted price to 2 decimal places. Then, these calculated discounted prices are pushed into a `discounted` array, which will then be returned by the function. Thus, giving us `[50, 100, 150]` for the call `discountPrices([100, 200, 300], .5)`. The array `discounted` will be accessible at Line 14 because `discounted` was intialized using `let` at Line 2, meaning it's scope is the entiriety of the function `discountPrices`, including in the for loop. 
9) At Line 11, a reference error will occur because `i` was initialized using `let` in a for loop block. As a result, the scope of `i` is confined to the for loop block and cannot be accessed outside of it. So, at Line 11 the reference error will state that it cannot find the variable `i`.
10) At Line 12, a reference error will occur because `discountedPrice` was initialized using `const` in a for loop block. As a result, the scope of `discountedPrice` is confined to the for loop block and cannot be accessed outside of it. So, at Line 12 the reference error will state that it cannot find the variable `discountedPrice`. 
11) An error will occur due to `finalPrice` being initialized with `const` and in the for loop, there is a line trying to set this const variable's value to another value. When a variable is initialized with `const`, its value cannot be changed. However in the for loop this being attempted, as a result this will cause a error due to trying to change the value of a `const` variable.
12) On the call `discountPrices([100, 200, 300], .5)` the function will not return any values because an error will occur. This is because `discounted` and `finalPrice` are both initialized using `const`, however their values are attempted to being changed in the following for loop. This will cause an error stating that this variable cannot be reassigned. 
13)
* a) `student.name`
* b) `student['Grad Year']`
* c) `




